qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -1516,3 +1527,17 @@ def validate(self, restriction_dict):
                 "columns:\n\t%s.\nSee the Templates tutorial for a description"
                 " of these fields." % ";\n\t".join(warning_msg),
                 qdb.exceptions.QiitaDBWarning)
+
+    def unique_columns(self):
+        """Get the unique columns in a template object
+
+        Returns
+        -------
+        set of str
+            Returns a set of the columns that are unique in a template object.
+        """
+
+        df = self.to_dataframe()
+
+        n = len(df)
+        return {k for k, v in df.nunique().items() if v == n}
```

Comment:
items -> iteritems cause items doesn't exist in the pandas version we have in qiita